app/controllers/folders_controller.rb

```diff
@@ -11,9 +12,13 @@ def create
     @folder = Folder.new(folder_params)
     if @folder.save
       flash[:success] = "#{@folder.name} Successfully Created"
-      redirect_to home_path
+      if current_folder
```

Comment:
@cmacaulay I agree with your assertion, in this case the second if/else statement is not needed.